fintech-examples/fintech-example-tests/src/test/java/de/adorsys/fintech/tests/e2e/FintechConsentUiSmokeE2ETest.java

```diff
@@ -0,0 +1,245 @@
+package de.adorsys.fintech.tests.e2e;
+
+import com.tngtech.jgiven.integration.spring.JGivenStage;
+import com.tngtech.jgiven.integration.spring.junit5.SpringScenarioTest;
+import de.adorsys.fintech.tests.e2e.config.ConsentAuthApproachState;
+import de.adorsys.fintech.tests.e2e.config.SmokeConfig;
+import de.adorsys.fintech.tests.e2e.steps.FintechServer;
+import de.adorsys.fintech.tests.e2e.steps.UserInformationResult;
+import de.adorsys.fintech.tests.e2e.steps.WebDriverBasedUserInfoFintech;
+import io.github.bonigarcia.seljup.SeleniumExtension;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.net.URI;
+
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.NONE;
+
+@JGivenStage
+@ActiveProfiles("test-mocked-fintech")
+@ExtendWith(SeleniumExtension.class)
+@SpringBootTest(classes = JGivenConfig.class, webEnvironment = NONE)
+public class FintechConsentUiSmokeE2ETest extends SpringScenarioTest<FintechServer, WebDriverBasedUserInfoFintech<? extends WebDriverBasedUserInfoFintech<?>>, UserInformationResult> {
+
+    @Autowired
+    private SmokeConfig smokeConfig;
+
+    @Autowired
+    private ConsentAuthApproachState state;
+
+    @BeforeEach
+    void memoizeConsentAuthorizationPreference() {
+        state.memoize();
+    }
+
+    @AfterEach
+    void restoreConsentAuthorizationPreference() {
+        state.restore();
+    }
+
+    @BeforeAll
+    static void setupDriverArch() {
+        WebDriverManager.firefoxdriver().arch64();
+    }
+
+    @Test
+    void testRedirectAntonBruecknerWantsToSeeItsAccountsAndTransanctionsFromFintech(FirefoxDriver firefoxDriver) {
```

Comment:
You should separate tests that validate account list from tests that validate transaction list. This is because you need to reuse the account list in future payments screen tests and it is always better to have granular tests per feature (because instead of going to stacktrace to see which method has failed and where it was in test case like it is in the mixed approach, you can clearly see that your accounts feature is working, but transactions feature is not)

Reply:
for the moment the test for accounts is normaly also for transactions because the account authorization implicitly authorizes the transactions as well.

Reply:
Yes and you can simply have:
```
private List<String> listAndAuthorizeAccountConsent() {
....
}

@Test
public void testAccounts() {
listAndAuthorizeAccountConsent();
}

@Test
public void testTransactions() {
List<String> accounts = listAndAuthorizeAccountConsent();
when()
  .click_on_account(...);
then()
  .transactions_are_received();
}
```

Reply:
ok i see thank you